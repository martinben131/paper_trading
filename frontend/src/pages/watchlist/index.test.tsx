import React from "react";
import { shallow } from "enzyme";
import Watchlist from "./index";

jest.mock("../../store", () => ({
  useProfile: () => [
    {
      watchlist: ["AAPL"],
    },
  ],
  useStocks: () => [
    {
      haveStocksBeenFetched: true,
      data: [
        {
          code: "AAPL",
          name: "Apple Inc.",
          priceCurr: 123.45,
          priceChange: 1.26,
        },
      ],
    },
  ],
}));

jest.mock("react-router-dom", () => ({
  useHistory: () => ({}),
}));

test("Renders watchlist page correctly", () => {
  const component = shallow(<Watchlist />);
  expect(component).toMatchSnapshot();
});
