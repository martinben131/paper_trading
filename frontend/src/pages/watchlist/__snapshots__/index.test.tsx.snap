// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders watchlist page correctly 1`] = `
<div
  className="stockContainer"
>
  <StockCard
    change={1.26}
    code="AAPL"
    key="AAPL"
    name="Apple Inc."
    price={123.45}
  />
</div>
`;
